### ISSUE TITLE
Better backup scheduling with additional triggers

### ISSUE THREAD

**Issue by a20eac1d** [kopia/kopia#3223]
I love Kopia, but I see a glaring issue that makes me (almost) not want to use it right now.

The way I understand it, Kopia only supports manual snapshots and timed snapshots.

The issue with manual snapshots is that you might forget to trigger them at the right time, or you simply want to have peace of mind that backups are taken care of automatically.

Timed backups are great in theory. Unfortunately, it seems like that Kopia is very rigid with the way they work. Let's say, for example, that I setup a trigger that creates a snapshot every day at 18:30 h. Now, if I shutdown my computer at 18:29, Kopia will not trigger a snapshot because the computer is turned off. And it also won't create a snapshot when the computer is turned back on, effectively "skipping" the timed snapshot all together. If this happens multiple day in a row, suddenly you've not done any snapshots for a few weeks even though they were correctly scheduled.

I understand the reasoning behind this, but I would like to suggest a change to this mechanic. I see two ways to fix this:

1. Add an option for additional triggers, such as "on user login" or "first time the computer is started today", etc.

or:

2. Make Kopia understand that the last scheduled snapshot was missed yesterday, and then create the snapshot immediately after startup

Please let me know what you think of this and how I can help. Thank you!

**Comment by lupusA** [comment]
Hi @a20eac1d, 

there is currently a bug for timed snapshots - especially for timed snaphshots with a specific time eg. 7 pm. If you setup a fixed period (e.g. every 12 hours), the missed snapshot is generated when the computer is back online again.

TLDR: There is currently a bug with fixed snapshots scheduled for a specific time rather than a period. 
See: #3197 

Cheers,

**Comment by julio-lopez** [comment]
Closing in favor of #3197
